backend/src/main/java/gov/cdc/usds/simplereport/config/ApiUserAwareGraphQlContextBuilder.java

```diff
@@ -99,7 +99,9 @@ private DataLoaderRegistry buildDataLoaderRegistry() {
         new DataLoader<>(
             personIds ->
                 supplyAsync(
-                    () -> patientPreferencesRepository.findAllByPersonInternalIdIn(personIds)));
+                    () ->
+                        patientPreferencesRepository.findAllAndCoalesceEmptyByPersonInternalIdIn(
```

Comment:
Looking at [BatchLoader](https://github.com/graphql-java/java-dataloader/blob/master/src/main/java/org/dataloader/BatchLoader.java), I think the contract of the interface is that the returned list must be index-matched to the input list. I can't see how we can possibly guarantee that with a database call (it might work, since postgresql's query plan is probably just to iterate over the list of inputs and do index lookups, but they certainly don't promise to return rows in a specific order). That being the case, I think that neither the original nor the new implementation is safe.

Reply:
Ah, you're right. Based on the error message, I expected that this was implemented with Maps under the hood, as opposed to an ordinal List. Your native Java implementation was actually my first inclination, but I was perturbed by its O(n^2) performance in application code and deferred to joining at the db level instead.

Due to pagination, though, the datasets that hit this query shouldn't be enormous – and you're entirely correct that ignoring the index order of the list is dangerous and we can't rely on postgres to return the rows in any particular order. 

Thanks for your feedback! I'll open a new PR with the application-layer implementation and scrap the native query.